core/app_shared.go

```diff
@@ -30,14 +32,27 @@ func (a appShared) getSurvey(id string, orgID string, appID string) (*model.Surv
 	return a.app.storage.GetSurvey(id, orgID, appID)
 }
 
-func (a appShared) getSurveys(orgID string, appID string, surveyIDs []string, surveyTypes []string, limit *int, offset *int) ([]model.Survey, error) {
-	return a.app.storage.GetSurveys(orgID, appID, surveyIDs, surveyTypes, limit, offset)
+func (a appShared) getSurveys(orgID string, appID string, surveyIDs []string, surveyTypes []string, calendarEventID string, limit *int, offset *int) ([]model.Survey, error) {
+	return a.app.storage.GetSurveys(orgID, appID, surveyIDs, surveyTypes, calendarEventID, limit, offset)
 }
 
 func (a appShared) createSurvey(survey model.Survey) (*model.Survey, error) {
 	survey.ID = uuid.NewString()
 	survey.DateCreated = time.Now().UTC()
 	survey.DateUpdated = nil
+
+	if len(survey.CalendarEventID) > 0 {
+		// check if user is admin of calendar event
+		user := calendar.User{AccountID: survey.CreatorID} // TODO: Add networkID
+		eventUsers, err := a.app.calendar.GetEventUsers(survey.OrgID, survey.AppID, survey.CalendarEventID, []calendar.User{user}, nil, "admin", nil)
+		if err != nil {
+			return nil, err
+		}
+		if len(eventUsers) == 0 { // user is not admin
```

Comment:
Since the Calendar BB API is currently not applying the provided filters we need to check to make sure that the current user is in the returned list and that they have the admin role.